### PR TITLE
Add blog for export logs to BigQuery,GCS,Pub/Sub through Stackdriver.

### PR DESCRIPTION
**Pull request by fanfanbj**
This blog is a guide to user about how to export Istio access logs to your favorite log sinks, such as BigQuery, GCS, Pub/Sub through Stackdriver.

### CODE REVIEW

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+[Bookinfo](/docs/examples/bookinfo/) 示例应用程序在整个任务中用作示例应用程序。
+
+## 开始前
+
+在集群中[安装 Istio](/docs/setup/) 并部署应用程序。
+
+## 配置 Istio 导出日志
+
+Istio 使用 `logentry` [模板](/docs/reference/config/policy-and-telemetry/templates/logentry)导出日志。 这里指定了可用于分析的所有变量。 它包含源服务、目标服务、auth指标（即将......）等信息。 以下是示意图：
```
Comment (loverto) [inline content_zh/blog/2018/export-logs-through-stackdriver/index.md]: auth指标-》auth 指标
少空格

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+{{< image width="75%" ratio="75%" link="./istio-analytics-using-stackdriver.png" caption="Diagram of exporting logs from Istio to Stackdriver for analysis" >}}
```
Comment (loverto) [inline content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 图片地址修改为绝对路径

Comment (loverto) [inline content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 翻译图片的标题

**Comment by rootsongjc** [comment]
@fanfanbj CI failed with the following output.
```
mdspell 1.3.1
mdl 0.5.0
htmlproofer 3.9.2
    content_zh/blog/2018/export-logs-through-stackdriver/index.md
       20 |  这里指定了可用于分析的所有变量。 它包含源服务、目标服务、auth指标（即将......）等信息。 以下是示意图： 
       32 | 2. 确保配置的接收器的 principalEmail 具有对项目写入权限和日志管理员角色的权限。 
       33 | 3. 确保已设置 GOOGLE_APPLICATION_CREDENTIALS 环境变量。 请按照[此处](https://cloud.g 
       40 | tform-setup/gke/) ，请确保在集群上启用了 bigquery [Scope](https://cloud.google. 
      151 | Bookinfo 示例，请使用浏览器访问 “http://$GATEWAY_URL/productpage“ 或发出以下命令： 
      151 | 请使用浏览器访问 “http://$GATEWAY_URL/productpage“ 或发出以下命令： 
      160 | ogle.com/) ，在接收器的数据集中找到一个前缀为 “accesslog_logentry_istio” 的表。 
      161 | 	*	GCS：导航到你的项目的 [Storage Brower](https://pantheon.corp.google 
      178 | 2.  在 spec 上增加 logInfo 
      219 |    如果您不打算任何后续任务，请参阅 [Bookinfo cleanup](/docs/examples/bookinfo/#cle 

>> 10 spelling errors found in 250 files
To learn how to address spelling errors, please see https://github.com/istio/istio.github.io#linting
content_zh/blog/2018/export-logs-through-stackdriver/index.md:152: MD009 Trailing spaces
content_zh/blog/2018/export-logs-through-stackdriver/index.md:168: MD009 Trailing spaces
content_zh/blog/2018/export-logs-through-stackdriver/index.md:195: MD009 Trailing spaces
content_zh/blog/2018/export-logs-through-stackdriver/index.md:159: MD010 Hard tabs
content_zh/blog/2018/export-logs-through-stackdriver/index.md:160: MD010 Hard tabs
content_zh/blog/2018/export-logs-through-stackdriver/index.md:161: MD010 Hard tabs
content_zh/blog/2018/export-logs-through-stackdriver/index.md:162: MD010 Hard tabs
content_zh/blog/2018/export-logs-through-stackdriver/index.md:194: MD010 Hard tabs
content_zh/blog/2018/export-logs-through-stackdriver/index.md:26: MD012 Multiple consecutive blank lines
content_zh/blog/2018/export-logs-through-stackdriver/index.md:32: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:33: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:38: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:39: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:40: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:45: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:46: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:51: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:52: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:53: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:137: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:150: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:157: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:178: MD029 Ordered list item prefix
content_zh/blog/2018/export-logs-through-stackdriver/index.md:153: MD031 Fenced code blocks should be surrounded by blank lines
content_zh/blog/2018/export-logs-through-stackdriver/index.md:151: MD034 Bare URL used

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
Running ["ScriptCheck", "HtmlCheck", "LinkCheck", "ImageCheck", "OpenGraphCheck"] on ["./public"] on *.html... 


Found 0 links in the cache...
Adding 502 links to the cache...
Removing 0 links from the cache...
Checking 502 external links...
Ran on 547 files!


HTML-Proofer finished successfully.
LINTING FAILED
Exited with code 1
```
Please make a commit to fix them.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md (hunk 2 above)
Comment (loverto) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 不要把静态文件放过来，静态文件请指向英文的静态资源

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md (hunk 1 above)
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

**Comment by fanfanbj** [comment]
@rootsongjc CI failing issue has been fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md (hunk 2 above)
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+1. 为项目启动 [`Stackdriver Monitoring API`](https://cloud.google.com/monitoring/api/enable-api) 。
+1. 确保配置的接收器的 `principalEmail` 具有对项目写入权限和日志管理员角色的权限。
+1. 确保已设置 `GOOGLE_APPLICATION_CREDENTIALS` 环境变量。请按照[`此处`](https://cloud.google.com/docs/authentication/getting-started)的说明进行设置。
+
+#### BigQuery
+
+1. [`创建 BigQuery 数据集`](https://cloud.google.com/bigquery/docs/datasets)作为日志导出的目标。
+1. 记录数据集的 ID 。 这里需要设置 Stackdriver 处理程序。 它的格式为`bigquery.googleapis.com/projects/[PROJECT_ID]/datasets/[DATASET_ID]`
+1. 给[`作者签名`](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination)：cloud-logs@system.gserviceaccount.com 。它具有 IAM 中的 BigQuery Data Editor 的角色。
```
Comment (fleeto) [inline content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 推测是这个意思，下同？进行[写入授权](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination)：在 IAM 中是 `cloud-logs@system.gserviceaccount.com`  BigQuery 数据编辑角色。

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+1.  保存如下的yaml文件为 `stackdriver.yaml` 。并替换 `<project_id>,
+    <sink_id>, <sink_destination>, <log_filter>` 为相应的值。
+
+    {{< text yaml >}}
+    apiVersion: "config.istio.io/v1alpha2"
+    kind: stackdriver
+    metadata:
+      name: handler
+      namespace: istio-system
+    spec:
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+#### Google Cloud Storage (GCS)
+
+1. [`创建 GCS 存储桶`](https://cloud.google.com/storage/docs/creating-buckets)，你希望在 GCS 中导出日志。
+1. 记录存储桶的 ID 。这里需要配置 Stackdriver 。它的形式为`storage.googleapis.com/ [BUCKET_ID]`。
+1. 给[`作者签名`](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination)：`cloud-logs @ system.gserviceaccount.com` 。它具有 IAM 中 Storage Object Creator 的角色。
+
+#### Google Cloud Pub/Sub
+
+1. [`创建主题`](https://cloud.google.com/pubsub/docs/admin)，你希望导出日志到Google Cloud Pub/Sub 中。
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+1. 验证日志是否正在通过 Stackdriver 流向配置的接收器。
+
+* Stackdriver：导航到你的项目的 [`Stackdriver Logs Viewer`](https://pantheon.corp.google.com/logs/viewer) ,查看 “GKE Container” -> “Cluster Name” -> “Namespace Id” , 查看 Istio 访问日志。
+* BigQuery：导航到你的项目的 [`BigQuery Interface`](https://bigquery.cloud.google.com/) ，在接收器的数据集中找到一个前缀为 `accesslog_logentry_istio` 的表。
+* GCS：导航到你的项目的 [`Storage Brower`](https://pantheon.corp.google.com/storage/browser/) ，在接收器的桶中找到一个名为 `accesslog.logentry.istio-system` 的桶。
+* Pub/Sub：导航到项目的 [`Pub/Sub 主题列表`](https://pantheon.corp.google.com/cloudpubsub/topicList) , 在接收器的主题中找到 `accesslog` 主题。
+
+## 了解发生了什么
+
+上面的 `Stackdriver.yaml` 文件配置了 Istio 将访问日志发送到 Stackdriver ，然后添加了一个接收器配置， 将日志导出。 具体如下：
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+        - destinationIp
+        ...
+        ...
+        sinkInfo:
+          id: '<sink_id>'
+          destination: '<sink_destination>'
+          filter: '<log_filter>'
+    {{< /text >}}
+
+在上面的配置中，sinkInfo包含有关日志导出到所需接收器的信息。 有关如何填写不同接收器的更多信息，请参阅[`此处`](https://cloud.google.com/logging/docs/export/#sink-terms)。
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+
+    {{< text bash >}}
+    $ kubectl delete -f stackdriver.yaml
+    {{< /text >}}
+
+*   如果您不打算任何后续任务，请参阅 [`Bookinfo cleanup`](/docs/examples/bookinfo/#cleanup) 指令关闭应用程序。
+
+## 日志导出的可用性
+
+导出到 BigQuery 只需几分钟（我们认为它几乎是即时的）， GCS 可以延迟2至12小时，Pub/Sub 几乎立即。
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+[`Bookinfo`](/docs/examples/bookinfo/) 示例应用程序在整个任务中用作示例应用程序。
+
+## 开始前
+
+在集群中[`安装 Istio`](/docs/setup/) 并部署应用程序。
+
+## 配置 Istio 导出日志
+
+Istio 使用 `logentry` [`模板`](/docs/reference/config/policy-and-telemetry/templates/logentry)导出日志。这里指定了可用于分析的所有变量。它包含源服务、目标服务、`auth` 指标（即将......）等信息。以下是示意图：
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md (hunk 3 above)
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 改成了给“接收器授权”。

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+## 清理
+
+*   删除新的 Stackdriver 配置：
+
+    {{< text bash >}}
+    $ kubectl delete -f stackdriver.yaml
+    {{< /text >}}
+
+*   如果您不打算任何后续任务，请参阅 [`Bookinfo cleanup`](/docs/examples/bookinfo/#cleanup) 指令关闭应用程序。
```
[the remark this answers is not in the document]
Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed all links to /zh/docs....

File: content_zh/blog/2018/export-logs-through-stackdriver/index.md [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+            destination: '<sink_destination>'
+            filter: '<log_filter>'
+    ---
+    apiVersion: "config.istio.io/v1alpha2"
+    kind: rule
+    metadata:
+      name: stackdriver
+      namespace: istio-system
+    spec:
+      match: "true" # If omitted match is true.
```
Comment (fleeto) [inline content_zh/blog/2018/export-logs-through-stackdriver/index.md]: 这里有注释

Comment (fanfanbj) [reply content_zh/blog/2018/export-logs-through-stackdriver/index.md]: fixed.

**Pull request merged**